astro/src/content/quickstarts/quickstart-go-api.mdx

```diff
@@ -0,0 +1,373 @@
+
+```json
+{
+  "aud": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "exp": 1689289585,
+  "iat": 1689285985,
+  "iss": "http://localhost:9011",
+  "sub": "00000000-0000-0000-0000-111111111111",
+  "jti": "ebaa4184-2320-47dd-925b-2e18756c635f",
+  "authenticationType": "PASSWORD",
+  "email": "teller@example.com",
+  "email_verified": true,
+  "applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "roles": [
+    "teller"
+  ],
+  "auth_time": 1689285985,
+  "tid": "d7d09513-a3f5-401c-9685-34ab6c552453"
+}
+```
+
+You need to validate the token and make sure the user has the appropriate role to access the API. You can do so by adding the following code:
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="authorization"
+  lang="go"/>
+
+There is a lot of code in this function. Let's break it down:
+- First you validate some claims. Notice the `checkAudience` that makes sure the token is intended for this application and `checkIss` to make sure the issuer is as expected.
+- `SetPublicKey`, which is a function we will create later, retrieves the public key for the JWT
+- After the key has been set, the `golang-jwt/jwt` will validate the signature and expiration date.
+- Once the token is validated, you will need to take a look at the roles.
+- The code will look to see which page is being called and then create an array for the roles that are valid for that request in `validRoles`.
+     - The containsRole function will be created as well and that creates an array with the intersection of roles that are allowed and roles that are required. If the length of that intersection is greater than 1 then the user has the proper role.
+- Once the roles have been validated, then the endpoint is served.
+
+Now, you will add the code to set the public key. This function will accept the key id from the token and call the authorization server to get the public key with that key id and put it in the proper format to be used.
+
+<Aside type="note">
+In this example, if the key is already set, it will not call it again. In a production environment you could store this key on the server or set it once on initialization. If your api serves several applications, you will need to get the correct public key for each application if they are different.
+</Aside>
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="set-public-key"
+  lang="go"/>
+
+Then add the code to compare the required roles to the user's roles based on the token.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="contains-role"
+  lang="go"/>
+
+Finally we will add a helper function that returns the name of the function being called by the endpoint.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="helper"
+  lang="go"/>
+
+That is all the code for the sample application. Save the `main.go` file.
```

Comment:
I had to install the jwt module:

```
go run main.go 
main.go:17:2: no required module provides package github.com/golang-jwt/jwt; to add it:
	go get github.com/golang-jwt/jwt
dan@mooreds your-application % go get github.com/golang-jwt/jwt
go: added github.com/golang-jwt/jwt v3.2.2+incompatible
```

Reply:
Did you?

Reply:
did you run `go get` before you tried `run main.go`? I believe that will get the module.

Reply:
I thought I did. I am running a slightly older version of golang, so perhaps that is the issue. Will try again and report back.

Reply:
It works great when I use go 1.21.